### PR TITLE
Disable HMPP

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: ktor-bom/build.gradle [hunk 1]
```diff
@@ -2,7 +2,8 @@ plugins {
     id 'java-platform'
 }
 
-apply from: rootProject.file('gradle/publish.gradle')
+// TODO
+// apply from: rootProject.file('gradle/publish.gradle')
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-bom/build.gradle]: thanks, fixed

File: ktor-client/ktor-client-tests/build.gradle.kts [hunk 2]
```diff
@@ -180,16 +180,16 @@ rootProject.allprojects {
             if (!(rootProject.ext.get("native_targets_enabled") as Boolean)) {
                 return@sourceSets
             }
-            if (KtorBuildProperties.ideaActive) {
+            if (IDEA_ACTIVE) {
                 if (name == "posixTest") {
                     getByName(name) {
                         dependencies {
-                            val hostname: String by project.ext
+//                            val hostname: String by project.ext
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/build.gradle.kts:187]: please uncomment or delete this code

File: ktor-client/ktor-client-tests/build.gradle.kts [hunk 3]
```diff
@@ -126,13 +126,13 @@ kotlin.sourceSets {
         } else {
             val posixTest by getting {
                 dependencies {
-                    val hostname: String by project.ext
+//                    val hostname: String by project.ext
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/build.gradle.kts:129]: same here

File: ktor-client/ktor-client-tests/build.gradle.kts (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/build.gradle.kts:187]: Thanks, fixed

File: ktor-client/ktor-client-tests/build.gradle.kts (hunk 3 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/build.gradle.kts:129]: Thanks, fixed!

**Comment by e5l** [comment]
Sure, will check it along with the code-style

File: buildSrc/src/main/kotlin/JvmConfig.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                    }
+
+                    api("org.slf4j:slf4j-api:$slf4j_version")
+                }
+            }
+
+            val jvmTest by getting {
+                dependencies {
+                    api("org.jetbrains.kotlin:kotlin-test")
+                    api("org.jetbrains.kotlin:kotlin-test-junit")
```
Comment (olme04) [inline buildSrc/src/main/kotlin/JvmConfig.kt:58]: junit artifacts added twice.
Also, does it make sense to add them, as now test dependencies added automatically when adding `kotlin-test` to common sourceset? same as with adding `stdlib` (in all targets)

File: buildSrc/src/main/kotlin/JvmConfig.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import org.jetbrains.kotlin.gradle.targets.jvm.tasks.*
+
+val JDK_16: String? = System.getProperty("JDK_16")
+val JDK_17: String? = System.getProperty("JDK_17")
+val JDK_18: String? = System.getProperty("JDK_18")
+
+val JDK_11: String? = System.getProperty("JDK_11")
+
+fun Project.configureJvm() {
+    val jdk = when (name) {
```
Comment (olme04) [inline buildSrc/src/main/kotlin/JvmConfig.kt:17]: `KtorBuildProperties.projectJdk` do the same (duplicate code)

File: buildSrc/src/main/kotlin/TargetsConfig.kt [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    val hasDesktop = hasPosix || files.any { it.name == "desktop" }
+    val hasNix = hasPosix || files.any { it.name == "nix" }
+    val hasDarwin = hasNix || files.any { it.name == "darwin" }
+    val hasJs = hasCommon || files.any { it.name == "js" }
+    val hasJvm = hasCommon || files.any { it.name == "jvm" }
+
+    kotlin {
+        if (hasJvm) {
+            jvm()
+            configureJvm()
```
Comment (olme04) [inline buildSrc/src/main/kotlin/TargetsConfig.kt:28]: `configureJvm` also installs `jvm` target. for consistency may be it's better to leave target installing in one place?

File: buildSrc/src/main/kotlin/TargetsConfig.kt [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        if (hasJs) {
+            js {
+                nodejs()
+                browser()
+            }
+
+            configureJs()
+        }
+
+        if (hasPosix) {
```
Comment (olme04) [inline buildSrc/src/main/kotlin/TargetsConfig.kt:40]: is it expected that those targets will be created even if `IDEA_ACTIVE` is false?

File: buildSrc/src/main/kotlin/TargetsConfig.kt [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+                    val desktopTest by creating
+                }
+            }
+        }
+
+        sourceSets {
+            if (hasCommon) {
+                val commonMain by getting {
+                    dependencies {
+                        api("org.jetbrains.kotlin:kotlin-stdlib-common")
```
Comment (olme04) [inline buildSrc/src/main/kotlin/TargetsConfig.kt:77]: same comment about stdlib/test dependencies as in `JvmConfig` file

File: buildSrc/src/main/kotlin/TargetsConfig.kt [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+                extra.set("commonStructure", true)
+            }
+            if (hasJvmAndNix) {
+                val jvmAndNixMain by creating {
+                    findByName("commonMain")?.let { dependsOn(it) }
+
+                    dependencies {
+                        api("org.jetbrains.kotlin:kotlin-stdlib-common")
+                        api("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version")
```
Comment (olme04) [inline buildSrc/src/main/kotlin/TargetsConfig.kt:96]: is it needed, as `common` already contains it?

File: buildSrc/src/main/kotlin/Versions.kt [hunk 10]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+val jdk8Modules = setOf(
```
Comment (olme04) [inline buildSrc/src/main/kotlin/Versions.kt:5]: same exists in `KtorBuildProperties` - is it really needed to duplicate it?

**Review by olme04 — COMMENTED** [review]
minor comments.
overall, Im just curios, why at all HMPP need to be disabled now, as soon HMPP will be enabled by default? :)

**Comment by e5l** [comment]
Hey @olme04, thanks for the review. I will take a look

File: buildSrc/src/main/kotlin/TargetsConfig.kt (hunk 7 above)
Comment (olme04) [reply buildSrc/src/main/kotlin/TargetsConfig.kt:40]: minor: I see that `posixTargets` also called down the file (so similar as with JVM double install of targets, even if should work, better to remove it from here for consistency, as this file is already hard to understand in detail

File: buildSrc/src/main/kotlin/JvmConfig.kt (hunk 4 above)
Comment (e5l) [reply buildSrc/src/main/kotlin/JvmConfig.kt:58]: Thank you, haven't noticed. Fixed

File: buildSrc/src/main/kotlin/TargetsConfig.kt (hunk 6 above)
Comment (e5l) [reply buildSrc/src/main/kotlin/TargetsConfig.kt:28]: Yep, it should work. Fixed

File: buildSrc/src/main/kotlin/TargetsConfig.kt (hunk 7 above)
Comment (e5l) [reply buildSrc/src/main/kotlin/TargetsConfig.kt:40]: `posixTargets()` is just an accessor POSIX targets(which create them if they don't exist).

I would say `posixTargets()` is called here for consistency: we create all targets before getting down to the source set configuration

File: buildSrc/src/main/kotlin/TargetsConfig.kt (hunk 9 above)
Comment (e5l) [reply buildSrc/src/main/kotlin/TargetsConfig.kt:96]: Thanks, fixed

File: buildSrc/src/main/kotlin/Versions.kt (hunk 10 above)
Comment (e5l) [reply buildSrc/src/main/kotlin/Versions.kt:5]: Thanks, removed

**Pull request merged**